src/components/mixins/SplashMixin.vue

```diff
@@ -133,52 +148,31 @@ export default class SplashMixin extends Vue {
                     'tsMods/updatePersistentCache',
                     {indexHash: packageListIndex.hash, chunks: packageListChunks}
                 );
+
+                this.loadingText = 'Processing the mod list';
+                this.getRequestItem('CacheOperations').setProgress(50);
+                return true;
             } catch (e) {
                 console.error('SplashMixin failed to cache mod list locally.', e);
             }
-
-            this.loadingText = 'Processing the mod list';
-        } else {
-            this.loadingText = 'Processing the mod list from cache';
         }
 
-        this.getRequestItem('CacheOperations').setProgress(50);
+        return false;
     }
 
     /***
      * Read mod list from the IndexedDB cache to Vuex so it's kept in memory.
-     * Always read from the IndexedDB since we don't know if the mod list was
-     * queried from the API successfully or not. This also handles the type
-     * casting, since mod manager expects the data to be formatted into objects.
-     *
-     * Failure at this point is no longer silently ignored, instead an error
-     * modal is shown.
-     *
-     * Return value is used to tell whether Vuex might contain an empty list
-     * after calling this because there was an error, or because the package
-     * list is actually empty.
      */
-    async readModsToVuex(): Promise<boolean> {
-        let isModListLoaded = false;
-
+    async readModsToVuex(): Promise<void> {
         try {
             await this.$store.dispatch('tsMods/updateMods');
-            isModListLoaded = true;
         } catch (e) {
-            this.$store.commit(
-                'error/handleError',
-                R2Error.fromThrownValue(e, `${this.loadingText} failed`)
-            );
+            console.error('SplashMixin failed to read mods to Vuex.', e);
```

Comment:
Following the above comment, the `read` here is misleading

Reply:
Comment updated.